packages/generator/src/generators/model-generator.ts

```diff
@@ -38,7 +39,26 @@ export class ModelGenerator extends Generator<ModelGeneratorOptions> {
       if (!this.options.dryRun) {
         // wrap in newlines to put a space below the previously generated model and
         // to preserve the EOF newline
-        this.fs.append(path.resolve("db/schema.prisma"), `\n${modelDefinition.toString()}\n`)
+        const schema = this.fs.read(path.resolve("db/schema.prisma"))
+
+        if (schema.indexOf(`model ${modelDefinition.name}`) === -1) {
+          //model does not exist in schema - just add it
+          this.fs.append(path.resolve("db/schema.prisma"), `\n${modelDefinition.toString()}\n`)
+        } else {
+          const model = matchBetween(schema, `model ${modelDefinition.name}`, "}")
+          if (model) {
+            //filter out all fields that are already defined
+            modelDefinition.fields = modelDefinition.fields.filter((field) => {
+              return model.indexOf(field.name) === -1
+            })
+
+            //add new fields to the selected model
+            const newModel = model.replace("}", `${modelDefinition.getNewFields()}}`)
+
+            //replace all content with the newly added fields for the already existing model
+            this.fs.write(path.resolve("db/schema.prisma"), schema.replace(model, newModel))
+          }
+        }
       }
       log.success(
```

Comment:
Awesome @engelkes-finstreet!! Works great!

One more tweak, let's switch this success message do say "created" or "updated" depending on whether it added the whole model or just updated it.